### PR TITLE
plugin: corrected support for basic actions for ardupilot

### PR DESCRIPTION
**Pull request by ykhedar**
At the moment, takeoff action for APM does not work at all with MAVSDK since takeoff command can only be executed with the flight mode Guided for APM. Arming works sometime but does not work after landing is completed. 

Following the work from https://github.com/mavlink/MAVSDK/pull/1703, this pull request tries to correct the broken actions (takeoff and arming) for Ardupilot. 

Following are the changes made:
1. Arming for Ardupilot(Copter 4.1.2) can be done only in specific flight modes: Hold, Stabilized and Offboard. To keep compatibility with PX4 and MAVSDK, Hold mode is selected for arming of the copter.
2. Takeoff can only be done in Offboard mode, hence a function specific to apm takeoff is implemented to take care of this.
3. Flight mode conversion between Arducopter and MAVSDK is also updated.
4. Integration tests are added/updated for mission raw and action hover. This way 7 more integration tests for APM will run now in addition to the 5 previously.

Looking forward to get reviews :)

### CODE REVIEW

**Review by JonasVautherin — CHANGES_REQUESTED** [review]
Nice! :rocket: 

A few suggestions, and I think you misspelled some `APM*` tests, I guess that's why it fails the PX4 CI checks

**Comment by julianoes** [comment]
@ykhedar I really appreciate your work here. I hope to get to it soon. That being said I also have a couple of refactoring changes in a local branch related to this, that will heavily conflict. I will have to resolve the conflicts.

**Review by JonasVautherin — CHANGES_REQUESTED** [review]

File: src/mavsdk/plugins/action/action_impl.cpp [hunk 1]
```diff
@@ -238,7 +238,8 @@ void ActionImpl::arm_async(const Action::ResultCallback& callback) const
     };
 
     if (_parent->get_flight_mode() == FlightMode::Mission ||
-        _parent->get_flight_mode() == FlightMode::ReturnToLaunch) {
+        _parent->get_flight_mode() == FlightMode::ReturnToLaunch ||
+        _parent->get_flight_mode() == FlightMode::Land) { // Needed for APM
```
[the remark this answers is not in the document]
Comment (ykhedar) [reply src/mavsdk/plugins/action/action_impl.cpp:242]: agree with your suggestion :)

File: src/mavsdk/core/system_impl.cpp [hunk 2]
```diff
@@ -847,21 +847,28 @@ ardupilot::CopterMode SystemImpl::flight_mode_to_ardupilot_copter_mode(FlightMod
...
         case FlightMode::Manual:
+            return ardupilot::CopterMode::Stabilize;
         case FlightMode::FollowMe:
+            return ardupilot::CopterMode::Follow;
         case FlightMode::Unknown:
-        case FlightMode::Ready:
-        case FlightMode::Takeoff:
+            return ardupilot::CopterMode::Unknown;
+        // case FlightMode::Ready:
+        // case FlightMode::Takeoff:
```
[the remark this answers is not in the document]
Comment (ykhedar) [reply src/mavsdk/core/system_impl.cpp]: moved to unknown for now.

**Comment by ykhedar** [comment]
@julianoes you mentioned before that you are working on a local branch which might conflict with this PR. Is that still the case? I did a rebase to the current main branch and resolved some conflicts thus arising.

**Review by JonasVautherin — APPROVED** [review]
Nice!

**Comment by ykhedar** [comment]
> Is the CI failure related to this PR?

Not really. Randomly one or two of the tests fail because the `health_all_ok` is not true. I did some digging and found its mostly because the home position not received/set. I did a test locally by making the home position request to retry every 2 seconds until home position is received something like this:

`_parent->add_call_every([this]() { request_home_position_again(); }, 2.0f, &_call_every_cookie);`

The change seems to make the integration tests pass (at least for the 10 times i tested locally.)
I was thinking making that another PR since was not related to this PR. If you disagree i can commit the changes to this PR

**Pull request merged**